.tekton/collector-slim-pull-request.yaml

```diff
@@ -27,7 +27,7 @@ spec:
   - name: image-expires-after
     value: '13w'
   - name: output-image-repo
-    value: quay.io/redhat-user-workloads/rh-acs-tenant/acs/collector-slim
```

Comment:
Can you please remind me, will `main`, `roxctl` and operator images think it is a release build in the sense of this code block here <https://github.com/stackrox/stackrox/blob/896452897dbc54bb60183782731dda95766e3a20/pkg/images/defaults/flavor.go#L103-L112>?
If yes, the repo is fine. If no, `-slim` should be part of the tag, not repo.

Reply:
TLDR: I think it's fine and the assumption that SCANNER_VERSION and COLLECTOR_VERSION tags will be generated should also hold.

Reply:
I think, the simplest for MVP^2 may be to switch the release off, push to `-latest`+`-slim`, and plan sorting out release in GA scope.

Reply:
Overriden with `GetAllVersionsDevelopment` (https://github.com/stackrox/stackrox/blob/master/pkg/images/defaults/flavor.go#L103, https://github.com/stackrox/stackrox/blob/master/pkg/version/version.go#L58, https://github.com/stackrox/stackrox/blob/master/pkg/version/internal/version_data.go#L7), so `<CollectorVersion>` as noted in `COLLECTOR_VERSION` file (I realized yesterday that we might have to `-fast` in that file). 

> -slim and -latest in suffix were introduced as a compatibility thing to not break old workflows. Maybe this is no longer necessary?

To be checked with collector team. I don't have a full understanding which tags they're pushing from their CI, looks like they're pushing full and slim to `quay.io/rhacs-eng/collector` incl `-slim` suffix: https://github.com/stackrox/collector/actions/runs/9095024517/job/24997429026#step:6:66, and `stackrox/stackrox` CI mirrors those images as `collector-slim:<MainVersion>` (https://github.com/stackrox/stackrox/blob/master/scripts/ci/lib.sh#L461-L462). So there are actually no images like `quay.io/rhacs-eng/collector-slim:<CollectorVersion>`. 

Is the `development_build` flavor && `releaseBuild` path unused? It should not work 🤔

Reply:
> I think, the simplest for MVP^2 may be to switch the release off, push to -latest+-slim, and plan sorting out release in GA scope.

IMO we can merge https://github.com/stackrox/stackrox/pull/10824, deploy a Central cluster and see what it would generate for collector(-slim). It might just work.

Reply:
> I realized yesterday that we might have to -fast in that file

We should not add suffixes to `COLLECTOR_VERSION` or `SCANNER_VERSION` because it can break builds outside of Konflux and/or product built outside of Konflux. We should look for other solution (overriding `BUILD_TAG` is my first candidate) - https://issues.redhat.com/browse/ROX-19958

> So there are actually no images like `quay.io/rhacs-eng/collector-slim:<CollectorVersion>`.

Correct!

> Is the development_build flavor && releaseBuild path unused?

I believe it executes for nightly builds and any other tagged builds.

> We can deploy a Central cluster and see what it would generate for collector(-slim).

There's a quicker way to check:

```
$ cd tmp
$ podman run --rm -it --entrypoint=/bin/bash -u 0:0 -v ./:/tmp quay.io/redhat-user-workloads/rh-acs-tenant/acs/roxctl:4.4.x-685-g7690ea6ed0-fast
$ roxctl helm output secured-cluster-services --output-dir /tmp/chart-konflux --remove --image-defaults=development_build
INFO:	Written Helm chart secured-cluster-services to directory "/tmp/chart-konflux"
$ exit
$ /assets/downloads/cli/roxctl-linux helm output secured-cluster-services --output-dir /tmp/chart-gha --remove --image-defaults=development_build
INFO:	Written Helm chart secured-cluster-services to directory "/tmp/chart-gha"
$ exit
```

Then, study `internal/defaults/50-images.yaml` in both dirs.

`/tmp/chart-konflux/internal/defaults/50-images.yaml` has:
```yaml
image:
  registry: quay.io/rhacs-eng
  main:
    name: main
    pullPolicy: IfNotPresent
  collector:
    name: collector
    slimName: collector-slim
# ...
image:
  collector:
    {{- if ._rox.collector.slimMode }}
    tag: "4.4.x-685-g7690ea6ed0"
    pullPolicy: IfNotPresent
    {{- else }}
    tag: "4.4.x-685-g7690ea6ed0"
    pullPolicy: Always
    {{- end }}
```

`/tmp/chart-gha/internal/defaults/50-images.yaml`:
```yaml
image:
  registry: quay.io/rhacs-eng
  main:
    name: main
    pullPolicy: IfNotPresent
  collector:
    name: collector
    slimName: collector
# ...
image:
  collector:
    {{- if ._rox.collector.slimMode }}
    tag: "3.18.x-221-g580b954a9a-slim"
    pullPolicy: IfNotPresent
    {{- else }}
    tag: "3.18.x-221-g580b954a9a-latest"
    pullPolicy: Always
    {{- end }}
```

Reply:
Did you notice that `scanner.tag` is also the `<MainVersion>`, not `<ScannerVersion>`? 

```yaml
  scanner:
    name: scanner-slim
    tag: 4.4.x-685-g7690ea6ed0
```

I have to think about the other points..

Reply:
Right, that's the "unified versions" case of release builds, I expected it (we came up with this code in the Maple team while working on that image flavor stuff that occurred after onboarding all containers to CPaaS).